test/producePublicFakeFactors.py

```diff
@@ -283,496 +387,59 @@
-                 file='{HOME}/public/Htautau/FakeRate/{VERSION}/pieces/uncertainties_QCD_W.root'.format(HOME=home,VERSION=version),
-                 object='uncertainties_W_MVis_MT_up',
-                 vars=['mvis', 'mt']
-             ),
-             comb.find('ff_w')
-         ]
-     )
-    }
-)
-comb_w_down = replace_nodes(
-    comb, 
-    {'ff_w':
-     Node(
-         name='ff_w_down',
-         formula='max(0.,1.-{sys_w_down})*{ff_w}',
-         leaves=[
-             Leaf(
-                 name='sys_w_down',
-                 file='{HOME}/public/Htautau/FakeRate/{VERSION}/pieces/uncertainties_QCD_W.root'.format(HOME=home,VERSION=version),
-                 object='uncertainties_W_MVis_MT_down',
-                 vars=['mvis', 'mt']
-             ),
-             comb.find('ff_w')
-         ]
-     )
-    }
-)
-comb_tt_corr_up = replace_nodes(
-    comb, 
-    {'ff_tt':
-     Node(
-         name='ff_tt_corr_up',
-         formula='(1.+{sys_tt})*{ff_tt}',
-         leaves=[
-             Leaf(
-                 name='sys_tt',
-                 file='{HOME}/public/Htautau/FakeRate/{VERSION}/pieces/sys_smooth_tt_nonclosure_mvis.root'.format(HOME=home,VERSION=version),
-                 object='sys_smoothed',
-                 vars=['mvis']
-             ),
-             comb.find('ff_tt')
-         ]
-     )
-    }
-)
-comb_tt_corr_down = replace_nodes(
-    comb, 
-    {'ff_tt':
-     Node(
-         name='ff_tt_corr_down',
-         formula='max(0.,1.-{sys_tt})*{ff_tt}',
-         leaves=[
-             comb_tt_corr_up.find('sys_tt'),
-             comb.find('ff_tt')
-         ]
-     )
-    }
-)
-comb_tt_stat_up = replace_nodes(
-    comb, 
-    {'ff_tt':
-     Node(
-         name='ff_tt_stat_up',
-         formula='(1.+{stat_tt})*{ff_tt}',
-         leaves=[
-             Leaf(
-                 name='stat_tt',
-                 file='{HOME}/public/Htautau/FakeRate/{VERSION}/pieces/toyerr_smooth_tt_mvis.root'.format(HOME=home,VERSION=version),
-                 object='hh_t_mvis_smoothed',
-                 vars=['mvis']
-             ),
-             comb.find('ff_tt')
-         ]
-     )
-    }
-)
-comb_tt_stat_down = replace_nodes(
-    comb, 
-    {'ff_tt':
-     Node(
-         name='ff_tt_stat_down',
-         formula='max(0.,1.-{stat_tt})*{ff_tt}',
-         leaves=[
-             comb_tt_stat_up.find('stat_tt'),
-             comb.find('ff_tt')
-         ]
-     )
-    }
-)
-
-
-## Fill fake factors
-fill(ff_qcd_ss, qcd_ss)
-fill(ff_qcd_ss, qcd_ss_up,   sys='ff_qcd_up')
-fill(ff_qcd_ss, qcd_ss_down, sys='ff_qcd_down')
 fill(ff_qcd_os, qcd_os)
-fill(ff_qcd_os, qcd_os_up,   sys='ff_qcd_up')
-fill(ff_qcd_os, qcd_os_down, sys='ff_qcd_down')
+fill(ff_qcd_os, qcd_os_up,   sys='ff_qcd_syst_up')
+fill(ff_qcd_os, qcd_os_down, sys='ff_qcd_syst_down')
+fill(ff_qcd_os, qcd_os_up_stat,   sys='ff_qcd_stat_up')
+fill(ff_qcd_os, qcd_os_down_stat, sys='ff_qcd_stat_down')
+fill(ff_qcd_os, frac_qcd, sys='frac_qcd')
```

Comment:
What is this systematic? I don't understand why you are replacing ff_qcd_os with {frac_qcd}+0*{ff_qcd_os} (so = frac_qcd).